centralized-sampling-tests/sample-apps/spring-boot/src/main/java/com/amazon/sampleapp/Application.java

```diff
@@ -0,0 +1,70 @@
+package com.amazon.sampleapp;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.propagation.W3CTraceContextPropagator;
+import io.opentelemetry.context.propagation.ContextPropagators;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.contrib.awsxray.AwsXrayIdGenerator;
+import io.opentelemetry.contrib.awsxray.AwsXrayRemoteSampler;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter;
+import io.opentelemetry.extension.aws.AwsXrayPropagator;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+
+@SpringBootApplication
+public class Application {
+
+	public static Resource resource = Resource.builder().build();
+
+	// Opentelemetry builder to create a xray remote sampler with polling interval of 1
+	public static OpenTelemetry openTelemetry =
+			OpenTelemetrySdk.builder()
+					.setPropagators(
+							ContextPropagators.create(
+									TextMapPropagator.composite(
+											W3CTraceContextPropagator.getInstance(), AwsXrayPropagator.getInstance())))
+					.setTracerProvider(
+							SdkTracerProvider.builder()
+									.addSpanProcessor(
+											BatchSpanProcessor.builder(OtlpGrpcSpanExporter.getDefault()).build())
+									.setResource(resource)
+									.setSampler(
+											AwsXrayRemoteSampler.newBuilder(resource)
+													.setPollingInterval(Duration.ofSeconds(1))
+													.build())
+									.setIdGenerator(AwsXrayIdGenerator.getInstance())
+									.build())
+					.buildAndRegisterGlobal();
+
+	public static void main(String[] args) {
+		// listenAddress should consist host + port (e.g. 127.0.0.1:5000)
+		String port;
+		String host;
+		String listenAddress = System.getenv("LISTEN_ADDRESS");
```

Comment:
where is `LISTEN_ADDRESS` from?

Reply:
it's from the java instrumentation sample app examples. It makes the host address configureable, so if I didn't want to use local host I could set the listen address as an environment variable. Could delete since it's not totally necessary but just extra configuration available in the sample app